tests/ulm-contracts/test_helpers.rs

```diff
@@ -14,3 +14,15 @@ fn decode_single_u256(bytes_id: u64) -> u256 {
   };
   value
 }
+
+fn decode_single_str(bytes_id: u64) -> str {
+  let (remaining_id, prefix_size) = :: bytes_hooks :: decode_u256(bytes_id);
+  if prefix_size != 32_u256 {
+      fail();
+  };
```

Comment:
It's more complicated. See this for more details: https://docs.soliditylang.org/en/latest/abi-spec.html but, as I understand it, the summary is something like this:

First, values are encoded in two bytes objects (called the "prefix" and the "suffix"), which are concatenated at the end to produce the full encoding. Fixed length values (such as int256) are encoded fully in the prefix (their suffix part is empty). Variable length values (such as strings) have both prefix and suffix parts. The prefix part is equal to `the-final-length-of-the-prefix + the-length-of-the-suffix-before-the-current-value` (I assume that this makes it easy to find the start of the suffix part of the value). The suffix part contains the length of the value followed by the value bytes.

So, `prefix_size` here is the prefix encoding of the string, which, for a single encoded string, should be the total length of the prefix. It can only be `32`, i.e. the number of bytes needed for any fixed size int with at most 256 bytes.

 The `:: bytes_hooks :: decode_str` call below will receive the suffix, and it will read the length, then it will extract that many bytes, and so on.